### PR TITLE
Add module for image extraction

### PR DESCRIPTION
**Pull request by KomalBhadani27**
Added image_extractor module which:

- Validates whether a tag is image tag or not.
- If tag is image tag, extracts the tag and creates and returns the Image Content object.

### CODE REVIEW

File: extraction/image_extractor.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    def get_image_attributes(self, node, image):
+        """This method extracts all attributes for the image"""
+
+        image.img_url = node['src']
+        if node.has_attr('title'):
+            image.img_title = node['title']
+        image.img_height, image.img_width = self.get_size(node)
+        image.is_gif = image.img_url.endswith('.gif')
+
+    def validate_and_extract(self, node):
```
Comment (detel) [inline extraction/image_extractor.py:27]: unit test for this method.

File: extraction/image_extractor.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    """This class inherits IContentExtractor to extract Images"""
+
+    def get_size(self, node):
+        """Returns the size of the image"""
+
+        if node.has_attr('width') and node.has_attr('height'):
+            return node['width'], node['height']
+        return 0, 0
+
+    def get_image_attributes(self, node, image):
```
Comment (detel) [inline extraction/image_extractor.py:18]: Add _, and move it down.

File: extraction/image_extractor.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+creates and returns the Image object"""
+
+from extraction.data_models.image import Image
+from extraction.interface_content_extractor import IContentExtractor
+
+
+class ImageExtractor(IContentExtractor):
+    """This class inherits IContentExtractor to extract Images"""
+
+    def get_size(self, node):
```
Comment (detel) [inline extraction/image_extractor.py:11]: Add _, and move it down.

Comment (detel) [reply extraction/image_extractor.py:11]: I guess you should create extraction_utils, and move this method there.
```
def get_size(self, node):
        """Returns the size of the image"""
        if node.has_attr('width') and node.has_attr('height'):
            return node['width'], node['height']
        return 0, 0
```
I'm see this method second time.

File: extraction/image_extractor.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+    def validate_and_extract(self, node):
+        if node.name == 'img' and node.has_attr('src'):
+            image_content = Image()
+            self.get_image_attributes(node, image_content)
+            return image_content
+        if node.name == 'figure':
+            img_tag = node.find('img')
+            if img_tag and img_tag.has_attr('src'):
+                image_content = Image()
```
Comment (detel) [inline extraction/image_extractor.py:35]: Its better to parameterised constructor with agruments as all element set in get_image_attributes

File: extraction/extractor.py [hunk 5]
```diff
@@ -77,16 +78,20 @@ def __extract_data_from_html_body(self, node):
...
         """This function will extract valid content and return it"""
 
-        # This function will be updated in next iteration
-        _ = node
         _ = self.soup
+        # checks if content is image
+        content_extractor = image_extractor.ImageExtractor()
+        image_content = content_extractor.validate_and_extract(node)
+        if image_content:
+            return image_content
```
Comment (detel) [inline extraction/extractor.py:95]: Create a list of all IContentExtractors and loop around it.

File: extraction/image_extractor.py (hunk 4 above)
Comment (detel) [reply extraction/image_extractor.py:35]: Update `get_image_attributes` to `create_image`. `create_image` can return Image(with, all, the, arguments)

File: extraction/image_extractor.py (hunk 1 above)
Comment (KomalBhadani27) [reply extraction/image_extractor.py:12]: Added!

File: extraction/image_extractor.py (hunk 2 above)
Comment (KomalBhadani27) [reply extraction/image_extractor.py]: Converted to create_image and moved down.

File: extraction/image_extractor.py (hunk 3 above)
Comment (KomalBhadani27) [reply extraction/image_extractor.py]: Moved to extraction_utils. Thanks for the suggestion.

File: extraction/image_extractor.py (hunk 4 above)
Comment (KomalBhadani27) [reply extraction/image_extractor.py]: Updated accordingly! Thanks for the suggestions.

File: extraction/extractor.py (hunk 5 above)
Comment (KomalBhadani27) [reply extraction/extractor.py]: Updated!

File: tests/test_extraction/test_image_extractor.py [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+       expected image content"""
+
+    assert isinstance(actual_image, Image)
+    assert actual_image.img_url == expected_image.img_url
+    assert actual_image.img_width == expected_image.img_width
+    assert actual_image.img_height == expected_image.img_height
+    assert actual_image.img_title == expected_image.img_title
+    assert actual_image.img_caption == expected_image.img_caption
+    assert actual_image.img_type == expected_image.img_type
+    assert actual_image.is_gif == expected_image.is_gif
```
Comment (detel) [inline tests/test_extraction/test_image_extractor.py:65]: Create a method for this

Comment (KomalBhadani27) [reply tests/test_extraction/test_image_extractor.py:65]: This is a custom assert method only which will be called from the test method.

Comment (detel) [reply tests/test_extraction/test_image_extractor.py:65]: My bad! You can move private methods to the bottom.

Comment (KomalBhadani27) [reply tests/test_extraction/test_image_extractor.py:65]: No problem! And I will move it to bottom.

**Pull request merged**